### ISSUE TITLE
Lint/ToEnumArguments fails on other methods

### ISSUE THREAD

**Issue by jacobat** [rubocop-hq/rubocop#9019]
The Lint/ToEnumArguments is supposed to warn when you're using `to_enum`/`enum_for` on the current method and miss arguments. It currently does not restrict itself to that and warns on other methods as well.

--------

## Expected behavior

No warning.

## Actual behavior

```
test.rb:10:5: W: Lint/ToEnumArguments: Ensure you correctly provided all the arguments.
    enum_for(:yieldr)
    ^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Create `test.rb`:
```
# frozen_string_literal: true

class F
  def yieldr(&blk)
    blk.call(1)
    blk.call(2)
  end

  def yieldr_enum
    enum_for(:yieldr)
  end
end

f = F.new
f.yieldr_enum.map { |n| n }
```

And then run:

```
rubocop test.rb
```

## RuboCop version

```
1.2.0 (using Parser 2.7.2.0, rubocop-ast 1.1.1, running on ruby 2.7.2 x86_64-darwin18)
  - rubocop-rails 2.8.1
  - rubocop-rspec 2.0.0
```

**Comment by dvandersluis** [comment]
There's a test case for this: https://github.com/rubocop-hq/rubocop/blob/9ceb446f41c47e28201d9b601fd2ff572d20c1cb/spec/rubocop/cop/lint/to_enum_arguments_spec.rb#L107-L114

This was discussed in #8955 but I'm not sure anything was decided.

**Comment by marcandre** [comment]
This is how the above code should be written:

```ruby
class F
  def yieldr(&blk)
    return to_enum(:yieldr) unless block
    blk.call(1)
    blk.call(2)
  end
end

f = F.new
f.yieldr.map { |n| n }
```

I'm assuming that this is just an example an you have a good reason to actually capture the block, but otherwise this is the Ruby way:

```ruby
class F
  def yieldr
    return to_enum(:yieldr) unless block_given?
    yield 1
    yield 2
  end
end
```
